# Estadistica para negocios
## Modulo 4: Inferencia estadistica

En este módulo, revisaremos de forma práctica como realizar pruebas de hipótesis utilizando muestas de datos.

Para ello, volveremos a tomar el rol de analistas. La gerencia está interesada en caracterizar a sus vendedores empleados. Para ello, cuenta con la base `base_ventas.xlsx`, la cual contiene las siguientes variables de una muestra 473 trabajadores:

- `ventas`: Ventas del trabajador en el ultimo periodo (en miles de pesos)
- `capacitacion`: Variable igual a 1 si el trabajador esta capacitado
- `edad`: Edad en años
- `educ`: Escolaridad en años

Primero, cargaremos los modulos relevantes de esta sesion: `pandas`, `matplotlib` y la distribucion normal y T de Student desde `scipy.stats`:

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from scipy.stats import norm, t

Luego, abriremos la base de datos y la guardamos en un objeto llamado `df`:

In [2]:
df = pd.read_excel('base_ventas.xlsx')
df.describe()

ventas  capacitacion        edad        educ
count   473.000000    473.000000  473.000000  473.000000
mean   1634.861653      0.348837   42.987315   12.143763
std    1279.665906      0.477107    6.676852    1.207835
min      35.894160      0.000000   23.000000    8.000000
25%     596.091861      0.000000   38.000000   12.000000
50%    1291.804155      0.000000   42.000000   12.000000
75%    2378.664494      1.000000   48.000000   12.000000
max    4999.109503      1.000000   63.000000   19.000000

## Pruebas de hipotesis sobre la media

### Cola inferior

Supongamos que la empresa desea tener trabajadores de mayor edad en su plana. La meta es aumentar la edad promedio a al menos 40 años. Sin embargo, la empresa no cuenta con información sobre toda la plana, sino nuestra muestra de 473 trabajadores. Se sabe que la desviación estándar de la edad es igual a $\sigma=7$.

Para probar esto, usaremos una prueba de hipótesis de cola inferior. Las hipótesis nula y alternativa son:

$$H_0: \mu \geq 40 $$
$$H_A: \mu < 40 $$

Luego, requerimos la media muestral de la edad:

In [3]:
media_edad = df['edad'].mean()

Luego, se construye el estadistico de prueba:

$$ z = \frac{\bar{X}-40}{7/\sqrt(473)}$$

In [4]:
z = (media_edad-40)/(7/np.sqrt(473))
z

9.281401316981661

Este valor se compara con el valor crítico de la distribución normal al 95% de confianza de una cola. Esto puede ser visto en una tabla (-1.645), o a través de Python, usando la distribución normal estándar (`norm()`) y el método `ppf()`, que entrega el valor de la distribución a una probabilidad determinada:

In [5]:
z_tabla = norm().ppf(0.05)
z_tabla

-1.6448536269514729

Es facil ver que el estadistico de prueba es **mayor** que el valor crítico de la distribución. Por lo tanto, **no se rechaza $H_0$**, por lo que no hay evidencia que pueda sustentar de que los trabajadores son menores de 40 años.

## Prueba de hipótesis con varianza desconocida

En la mayoria de los casos, es dificil contar con la varianza poblacional. Por esto, es necesario redefinir la prueba de hipotesis sobre la media, utilizando la **distribución T de Student**.

Supongamos que la empresa requiere saber si la escolaridad promedio de los trabajadores de la empresa es 12 años. Para ello, se pleantean las siguientes hipótesis:

$$H_0: \mu = 12 $$
$$H_A: \mu \neq 12 $$

Requerimos la media y la desviación estándar de la escolaridad:

In [6]:
media_esc = df['educ'].mean()
sd_esc = df['educ'].std()

print(media_esc,sd_esc)

12.143763217563105 1.2078346823845687


Luego, se construye el estadistico de prueba:

$$ T = \frac{\bar{X}-12}{s/\sqrt(473)}$$

In [7]:
T_stat = (media_esc-12)/(sd_esc/np.sqrt(473))
T_stat

2.588635236615983

Este valor se compara con el valor crítico de la distribución T de dos colas, con 472 grados de libertad, al 95% de confianza. Esto puede ser visto en una tabla (1.96), o a través de Python, usando la distribución normal estándar (`t()`) y el método `ppf()`, que entrega el valor de la distribución a una probabilidad determinada:

In [8]:
t_calculado = t(df=473).ppf(0.975)
t_calculado

1.9649919970066514

Es facil ver que el estadistico de prueba es **mayor** que el valor crítico de la distribución. Por lo tanto, **se rechaza $H_0$**, por lo que se puede asegurar que la escolaridad promedio es distinta de 12 años.

### Prueba de hipótesis sobre proporciones

Supongamos que la empresa busca saber si el nivel de trabajadores capacitados es superior al 40%. Para ello, se realizará una prueba de hipótesis de cola superior. Las hipótesis nula y alterna son:

$$H_0: p \geq 0.4 $$
$$H_A: p < 0.4 $$

La proporción muestral se obtiene directamente de la media de la variable de capacitación:

In [9]:
media_cap = df['capacitacion'].mean()
media_cap

0.3488372093023256

Luego, construimos el estadistico de prueba:

$$ z = \frac{\bar{p}-0.4}{\sqrt{(0.4(1-0.4))/473}}$$

In [10]:
z = (media_cap-0.4)/np.sqrt(0.4*(1-0.4)/473)
z

-2.271324443729291

Este valor se compara con el valor crítico de la distribución normal al 95% de confianza de una cola. Esto puede ser visto en una tabla (-1.645), o a través de Python, usando la distribución normal estándar (`norm()`) y el método `ppf()`, que entrega el valor de la distribución a una probabilidad determinada:

In [11]:
z_tabla = norm().ppf(0.05)
z_tabla

-1.6448536269514729

Es facil ver que el estadistico de prueba es **menor** que el valor crítico de la distribución. Por lo tanto, **se rechaza $H_0$**, por lo que se puede asegurar que la proporción de trabajadores capacitados es menor que 40%.

## Pruebas de hipótesis con dos muestras independientes

Supongamos que la empresa ahora está interesada en saber si hay una diferencia considerable de las ventas promedio entre quienes están capacitados y quienes no lo están. Para saber esto, la empresa sugiere realizar una prueba de diferencia de medias. Para esto, las hipótesis nula y alternativa son:

$$H_0: \mu_1 - \mu_2 = 0$$
$$H_A: \mu_1 - \mu_2 \neq 0$$

Donde $\mu_1$ es la media de las ventas de quienes están capacitados, y $\mu_2$ es la media de las ventas de quienes no lo están.

Primero, vamos a separar ambas muestras:

In [12]:
df1 = df[df['capacitacion'] == 1]
df2 = df[df['capacitacion'] == 0]

Luego, describiremos ambas bases de datos:

In [13]:
df1.describe()

ventas  capacitacion        edad        educ
count   165.000000         165.0  165.000000  165.000000
mean   1798.219761           1.0   41.569697   12.018182
std    1312.078064           0.0    6.357466    1.172514
min      51.472872           1.0   23.000000    8.000000
25%     737.963826           1.0   37.000000   12.000000
50%    1448.197324           1.0   41.000000   12.000000
75%    2608.277082           1.0   46.000000   12.000000
max    4902.619015           1.0   57.000000   16.000000

In [14]:
df2.describe()

ventas  capacitacion        edad        educ
count   308.000000         308.0  308.000000  308.000000
mean   1547.348381           0.0   43.746753   12.211039
std    1255.379486           0.0    6.729837    1.222912
min      35.894160           0.0   25.000000    9.000000
25%     540.223523           0.0   39.000000   12.000000
50%    1195.804903           0.0   44.000000   12.000000
75%    2208.895938           0.0   49.000000   12.000000
max    4999.109503           0.0   63.000000   19.000000

Las ventas promedio de quienes están capacitados son $1.798.220, mientras que las ventas promedio quienes no lo están es $1.547.348. Además, los capacitados son 165 individuos, mientras que los no capacitados son 308 individuos. Vamos a extraer esta información, además de las desviaciones estándar muestrales:

In [15]:
df1_media = df1['ventas'].mean()
df2_media = df2['ventas'].mean()

df1_sd = df1['ventas'].std()
df2_sd = df2['ventas'].std()

df1_n = len(df1)
df2_n = len(df2)

In [16]:
df1_n

165

Con esta información, podemos construir el estadistico de prueba:

$$ T = \frac{(\bar{X}_1-\bar{X}_2)-0}{\sqrt{\frac{s_1^2}{165} + \frac{s_2^2}{308}}}$$

In [17]:
T_stat = (df1_media - df2_media - 0)/np.sqrt(((df1_sd**2)/df1_n) + ((df2_sd**2)/df2_n))
T_stat

2.011776724495604

Los grados de libertad son:

$$gl = \frac{\left( \frac{s_1^2}{165} + \frac{s_2^2}{308} \right)^2}{\frac{1}{164}\left( \frac{s_1^2}{165} \right)^2 + \frac{1}{307} \left( \frac{s_2^2}{308} \right)}$$

In [18]:
gl = ((((df1_sd**2)/df1_n) + ((df2_sd**2)/df2_n))**2)/((1/(df1_n-1))*((df1_sd**2)/df1_n)**2 + (1/(df2_n-1))*((df2_sd**2)/df2_n)**2)
gl

322.82323117339956

Como el resultado no es un numero entero, se redondea hacia abajo. Para ello. usaremos la función `floor()` de `numpy`, y convertiremos el número a un entero (`int`):

In [19]:
gl = int(np.floor(gl))
gl

322

El estadistico de tabla T Student con 322 grados de libertad se aproxima a 1.96:

In [20]:
t_calculado = t(df=gl).ppf(0.975)
t_calculado

1.967358585322468

Es facil ver que el estadistico de prueba es **menor** que el valor crítico de la distribución. Por lo tanto, **se rechaza $H_0$**, por lo que se puede asegurar que las ventas promedio de los capacitados son considerablemente distintas a los de los no capacitados. Es más, debido a que la media de los capacitados es mayor, también se puede asegurar que **las ventas promedio de los capacitados son mayores que las de los no capacitados**.

### Diferencia de proporciones

Finalmente, supongamos que la empresa quiere saber si la proporción de personas capacitadas es mayor entre las personas con estudios superiores (más de 12 años de escolaridad) que quienes no tienen estudios superiores. Para ello, se puede utilizar la prueba de hipótesis para diferencia de proporciones. Las hipótesis nula y alternativa son:

$$H_0: (p_1 - p_2) = 0 $$
$$H_A: (p_1 - p_2) \neq 0 $$

Primero, vamos a separar ambas muestras de acuerdo a los años de escolaridad:

In [21]:
df1 = df[df['educ'] > 12]
df2 = df[df['educ'] <= 12]

Describiendo ambas muestras:

In [23]:
df1.describe()

ventas  capacitacion        edad        educ
count   109.000000    109.000000  109.000000  109.000000
mean   1480.609307      0.348624   42.541284   13.692661
std    1272.520458      0.478736    6.877947    1.159957
min      83.966237      0.000000   23.000000   12.100000
25%     456.310859      0.000000   38.000000   13.000000
50%    1101.683146      0.000000   42.000000   13.500000
75%    2166.362614      1.000000   47.000000   14.000000
max    4914.940393      1.000000   59.000000   19.000000

In [24]:
df2.describe()


ventas  capacitacion        edad        educ
count   364.000000    364.000000  364.000000  364.000000
mean   1681.052603      0.348901   43.120879   11.679945
std    1279.920912      0.477279    6.619256    0.748669
min      35.894160      0.000000   25.000000    8.000000
25%     612.294158      0.000000   38.000000   12.000000
50%    1350.782846      0.000000   42.500000   12.000000
75%    2482.408522      1.000000   48.000000   12.000000
max    4999.109503      1.000000   63.000000   12.000000

La proporción de capacitados es bastante cercana entre ambas muestras. El estadistico de prueba de diferencia de proporciones es:

$$z = \frac{(\bar{p_1}-\bar{p_2})}{\sqrt{\bar{p}(1-\bar{p})\left( \frac{1}{N_1} \frac{1}{N_2} \right)}} \sim N(0,1)$$

Donde $p_1$ es la proporción de capacitados para quienes tienen educación superior, y $p_2$ es la proporción de capacitados para quienes no tienen educación superior. Notar que $\bar{p}$ no es conocido, pero se puede estimar como:

$$\bar{p} = \frac{N_1 \bar{p_1} + N_2 \bar{p_2}}{N_1 + N_2}$$

In [22]:
df1_prop = df1['capacitacion'].mean()
df2_prop = df2['capacitacion'].mean()

df1_n = len(df1)
df2_n = len(df2)

Luego, calculamos el estimador de $p$:

In [25]:
p_est = (df1_prop * df1_n + df2_prop * df2_n)/(df1_n + df2_n)
p_est

0.3488372093023256

Ahora, claculamos el estadistico de prueba:

In [26]:
z = (df1_prop - df2_prop)/(p_est*(1-p_est)*(1/df1_n + 1/df2_n))
z

-0.10238095238093846

Sabemos que el valor critico de la distribucion normal estándar al 95% de confianza es 1.96:

In [27]:
z_tabla = norm().ppf(0.975)
z_tabla

1.959963984540054

Como el valor calculado del estadistico (en valor absoluto) es **menor** que el valor crítico, entonces **no se rechaza** la hipótesis nula. Por lo tanto, no existe evidencia para asegurar que el porcentaje de capacitados entre quienes tienen educación superior es distinto al porcentaje de capacitados entre quienes no tienen educación superior.